File: 3641.java
//* @author: 82638882@163.com
import java.util.Scanner;
public class Main
{
 public static void main(String[] args)
 {
  Scanner in=new Scanner(System.in);
  while(true)
  {
    long p=in.nextLong();
    long a=in.nextLong();
    if(a==0&&p==0) break;
    if(isPrime(p)){
        System.out.println("no");
        continue;
    }
    System.out.println(sum(a,p,p)==a?"yes":"no");
   }
  }

 public static boolean isPrime(long a)
 {
    if(a==2) return true;
    for(int i=3;i<=Math.sqrt(a);i+=2)
        if(a%i==0) return false;
    return true;
  }

 public static long sum(long a,long n,long p)
 {
    if(n==0) return 1;
    long w=sum(a*a%p,n/2,p);
    if(n%2==1) w=(w*a)%p;
    return w;
 }
}

File: 3641.cc
#include<iostream>
#include<cmath>
using namespace std;
int main()
{
	long long a,p;
	int isprime(long long n);
	long long modular(long long a,long long r,long long m);
	while(cin>>p>>a)
	{
		if(p==0&&a==0)
			break;
		long long result;
		if(isprime(p))
			cout<<"no"<<endl;
		else
		{
			if(a==modular(a,p,p))
				cout<<"yes"<<endl;
			else
				cout<<"no"<<endl;
		}
	}
	return 0;
}
int isprime(long long n)
{
	if(n==2)
		return 1;
	if(n<=1||n%2==0)
		return 0;
	long long j=3;
	while(j<=(long long)sqrt(double(n)))
	{
		if(n%j==0)
			return 0;
		j+=2;
	}
	return 1;
}
long long modular(long long a,long long r,long long m)
{
	long long d=1,t=a;
	while(r>0)
	{
		if(r%2==1)
			d=(d*t)%m;
		r/=2;
		t=t*t%m;
	}
	return d;
}



